src/webgpu/api/validation/render_pipeline/resource_compatibility.spec.ts

```diff
@@ -48,6 +48,12 @@ g.test('resource_compatibility')
         !t.hasLanguageFeature('readonly_and_readwrite_storage_textures'),
       'Storage textures require language feature'
     );
+    t.skipIf(
+      t.params.stage === 'vertex' &&
+        ((wgslResource.buffer !== undefined && wgslResource.buffer.type === 'storage') ||
+          wgslResource.storageTexture !== undefined),
+      'Storage buffers and textures cannot be used in vertex shaders'
+    );
```

Comment:
This seems too broad for textures. It should mirror the filter above and consider the access mode. Read-only storage texture should be permitted in vertex shaders. The skip above should already handle the language feature aspect.

Reply:
From the new line that was added to the spec, I thought the non-read-only limitation was for buffers only, should it be both?

> Storage buffers with an access mode other than read and storage textures cannot be statically accessed in a vertex shader stage.

https://www.w3.org/TR/WGSL/#var-and-value

Reply:
That seems like a spec bug. It is clearer in https://gpuweb.github.io/gpuweb/wgsl/#address-space.
> [Variables](https://gpuweb.github.io/gpuweb/wgsl/#variable) in the [storage](https://gpuweb.github.io/gpuweb/wgsl/#address-spaces-storage) address space ([storage buffers](https://gpuweb.github.io/gpuweb/wgsl/#storage-buffer)) can only be [statically accessed](https://gpuweb.github.io/gpuweb/wgsl/#statically-accessed) by a [vertex shader stage](https://gpuweb.github.io/gpuweb/wgsl/#vertex-shader-stage) if the access mode is [read](https://gpuweb.github.io/gpuweb/wgsl/#access-read). Variables whose [store type](https://gpuweb.github.io/gpuweb/wgsl/#store-type) is a [storage texture](https://gpuweb.github.io/gpuweb/wgsl/#type-storage-texture) with a [write](https://gpuweb.github.io/gpuweb/wgsl/#access-write) or [read_write](https://gpuweb.github.io/gpuweb/wgsl/#access-read_write) [access mode](https://gpuweb.github.io/gpuweb/wgsl/#access-mode) cannot be [statically accessed](https://gpuweb.github.io/gpuweb/wgsl/#statically-accessed) by a [vertex shader stage](https://gpuweb.github.io/gpuweb/wgsl/#vertex-shader-stage). See WebGPU [createBindGroupLayout()](https://gpuweb.github.io/gpuweb/#dom-gpudevice-createbindgrouplayout).

Reply:
Got it, thanks! I'll do a follow up fix.

Reply:
https://github.com/gpuweb/gpuweb/pull/4684 makes it consistent